### PR TITLE
Refactor the code to make it easier to support multiple db backends in the future.

### PR DESCRIPTION
**Pull request by grao1991**
Restructure the directories.
Rename several variables to remove "Pg".
Remove some unused methods.

Tested on the local machine, verified both manually running migration and running migration through rust code can work.

### CODE REVIEW

**Review by bowenyang007 — DISMISSED** [review]
There's a problem. Running migrations manually the old way anymore. We have to cd into the new folder to run migrations. It also generates schema in a different location. Can you actually try it and update the readme? Also please add test plan.

**Comment by grao1991** [comment]
> There's a problem. Running migrations manually the old way anymore. We have to cd into the new folder to run migrations. It also generates schema in a different location. Can you actually try it and update the readme? Also please add test plan.

Tested and added one line in README.

**Review by CapCap — APPROVED** [review]

File: rust/processor/src/db/common/mod.rs
Comment (CapCap) [inline rust/processor/src/db/common/mod.rs:1]: please make this `pub`

File: rust/processor/src/db/mod.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+pub(crate) mod common;
```
Comment (CapCap) [inline rust/processor/src/db/mod.rs:1]: pub again, no? otherwise how will SDK use any of this

File: rust/processor/src/db/common/mod.rs
Comment (grao1991) [reply rust/processor/src/db/common/mod.rs:1]: Done

File: rust/processor/src/db/mod.rs (hunk 1 above)
Comment (grao1991) [reply rust/processor/src/db/mod.rs]: Done

File: rust/processor/src/db/common/mod.rs
Comment (rtso) [reply rust/processor/src/db/common/mod.rs:1]: The command should be run inside `src/db/postgres/`

**Pull request merged**